vars/generateAndExecutePipeline.groovy

```diff
@@ -119,13 +134,56 @@ def _getApplicationDefinitionFilePath() {
     return 'applicationDefinition.yaml'
 }
 
+def _validateReleaseBranch(applicationDefinition, scmMetadata) {
+
+    if (isReleaseBranch()==true) {
+
+        def majorVersion = applicationDefinition.Pipeline.MajorVersion
+        def archetype    = applicationDefinition.Pipeline.Archetype
+
+        _assertMajorVersionMatchesReleaseVersion(majorVersion)
+
+    }
+}
+
+def _assertMajorVersionMatchesReleaseVersion(majorVersion) {
+
+    relBranchNm = env.BRANCH_NAME.split('/')[-1]
+
+    if (majorVersion.toString() != relBranchNm) {
+
+        error("Major version in application definition, ${majorVersion}, does not match release branch name: ${relBranchNm}")
+    }
+}
+
 def _getPipelineDefinition(applicationDefinition) {
 
     // do cool stuff here
 
     return pipelineDefinition
 }
 
+def _isReleaseOrPr() {
+
+    def isRelease    = isRelease()
+    def isPr         = _isPr()
+
+    return isRelease || isPr
+}
+
+def _isPr() {
+
+    return env.BRANCH_NAME ==~ /PR-\d+/
+}
+
+def _isFeature() {
+
+    isTruFals = env.BRANCH_NAME ==~ /feature\/.+/
```

Comment:
This can be simplified to one line.  Similar to `_isPr()`

```
def _isFeature() {

    return env.BRANCH_NAME ==~ /feature\/.+/
}
```

Reply:
That _would_ get rid of that pesky typo.

Reply:
Agreed. Will fix